inference-engine/src/mkldnn_plugin/nodes/mkldnn_depth_to_space_node.cpp

```diff
@@ -150,20 +154,16 @@ void MKLDNNDepthToSpaceNode::createPrimitive() {
 }
 
 void MKLDNNDepthToSpaceNode::prepareParams() {
-    execPtr = std::make_shared<DepthToSpaceExecutor>(attrs, getParentEdgeAt(0)->getMemoryPtr(), getChildEdgeAt(0)->getMemoryPtr());
+    attrs.srcBlockedDims = getParentEdgeAt(0)->getMemoryPtr()->GetDescWithType<BlockedMemoryDesc>()->getBlockDims();
+    execPtr = std::make_shared<DepthToSpaceExecutor>(attrs);
 }
 
-MKLDNNDepthToSpaceNode::DepthToSpaceExecutor::DepthToSpaceExecutor(const DepthToSpaceAttrs& attrs,
-                                                                   const MKLDNNMemoryPtr& srcMemPtr,
-                                                                   const MKLDNNMemoryPtr& dstMemPtr) {
-    VectorDims srcDims = srcMemPtr->getStaticDims();
-    VectorDims dstDims = dstMemPtr->getStaticDims();
-    size_t nDims = srcDims.size();
-    const size_t nSpatialDims = nDims - 2;
-
-    const bool isBlocked = srcMemPtr->getDesc().hasLayoutType(LayoutType::nCsp8c) ||
-                           srcMemPtr->getDesc().hasLayoutType(LayoutType::nCsp16c);
-    const size_t reshapedRank = nDims + nSpatialDims + static_cast<int>(isBlocked) + static_cast<int>(isBlocked && attrs.mode == Mode::DEPTH_FIRST);
+MKLDNNDepthToSpaceNode::DepthToSpaceExecutor::DepthToSpaceExecutor(const DepthToSpaceAttrs& attrs) {
+    size_t nDims = attrs.srcBlockedDims.size();
+    const bool isBlocked = attrs.layoutType == DepthToSpaceAttrs::Blocked;
+    const bool isChannelsFirst = attrs.layoutType == DepthToSpaceAttrs::ChannelsFirst;
+    const size_t nSpatialDims = nDims - 2 - static_cast<int>(isBlocked);
```

Comment:
This implementation supports blocked layout with only 1 block count. But anyway I rewrote this code part. Now `nSpatialDims` is initialized once in `createPrimitive`

Reply:
If the executor support only one block, should not we add some checks to it? Because the user can set CommonLayoutField to blocked an expect that any blocks number is supported.

Reply:
Changed logic as we discussed offline